babel/messages/extract.py

```diff
@@ -400,56 +401,65 @@ def extract(
                    options=options or {})
 
     for lineno, funcname, messages, comments in results:
-        spec = keywords[funcname] or (1,) if funcname else (1,)
+        specs = keywords[funcname] or (1,) if funcname else (1,)
+        if not isinstance(specs, dict): # for backwards compatibility
+            specs = {None: specs}
         if not isinstance(messages, (list, tuple)):
             messages = [messages]
         if not messages:
             continue
 
         # Validate the messages against the keyword's specification
-        context = None
-        msgs = []
-        invalid = False
-        # last_index is 1 based like the keyword spec
-        last_index = len(messages)
-        for index in spec:
-            if isinstance(index, tuple):
-                context = messages[index[0] - 1]
+
+        # None matches all arities.
+        for arity in (None, len(messages)):
+            try:
+                spec = specs[arity]
+            except KeyError:
+                continue
```

Comment:
Instead of this big huge for loop over the 2 arities, could the loop body here be broken out to a separate helper function?